src/adaPets.js

```diff
@@ -10,28 +10,56 @@ const BASE_URL = 'http://localhost:3000/pets/';
 // Option functions.
 const listPets = () => {
   // Fill out as part of Wave 1.
+  axios.get(BASE_URL)
+  .then((response) => {
+    setResult(response.data);
+  })
+  .catch((error) => {
+    // console.log(error.response);
+    setError("Unwanted result!")
+  });
 };
 
+
 const showDetails = (selectedPetId) => {
   if (!selectedPetId) {
-    setError("You tried to show details for a pet without selecting it!");
-    
+   setError("You tried to show details for a pet without selecting it!");
   }
-
+  
   // Fill out as part of Wave 2.
-};
+  axios.get(`${BASE_URL}${selectedPetId}`) 
+    .then((response) => {
+      setResult(response.data);
+    })
+    .catch((error) => { 
+      setError('You selected a show details option that failed and status is 404 now'); 
```

Comment:
You should include the actual error code here, not just assume it's 404:

```suggestion
      setError('You selected a show details option that failed and status is ${error.response.status} now'); 
```